src/control/server/mgmt_system.go

```diff
@@ -104,6 +105,87 @@ func getPeerListenAddr(ctx context.Context, listenAddrStr string) (net.Addr, err
 		net.JoinHostPort(tcpAddr.IP.String(), portStr))
 }
 
+const (
+	updateCheckTimeout = 1 * time.Second
+)
+
+func (svc *mgmtSvc) groupUpdateLoop(ctx context.Context) {
+	var updateRequested bool
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-time.After(updateCheckTimeout):
+			if !updateRequested {
+				continue
+			}
+			if err := svc.doGroupUpdate(ctx); err != nil {
+				svc.log.Error(errors.Wrap(err, "group update failed").Error())
+				if err == system.ErrEmptyGroupMap {
+					updateRequested = false
+				}
+				continue
```

Comment:
so , the behaviour indicated here is to retry every timout continuously if doGroupUpdate fails, I'm assuming that is intended? do we want to put a limit on the number of retries?

Reply:
The logic is that if at least one update request has been made, then we want to keep trying to perform a GroupUpdate until we succeed. If the GroupUpdate never succeeds, then the system must be in pretty bad shape.